docker/src/main/java/brooklyn/entity/container/docker/DockerHostImpl.java

```diff
@@ -692,6 +701,31 @@ public void postStart() {
         sensors().set(DOCKER_IMAGE_ID, imageId);
 
         scan = scanner();
+
+        Map<String, String> externalRegistries = config().get(DockerHost.DOCKER_EXTERNAL_REGISTRIES);
+        Map<String, String> externalRegistriesLoginDetails = config().get(DockerHost.DOCKER_EXTERNAL_REGISTRY_LOGIN_DETAILS);
+
+        if(externalRegistries != null && !externalRegistries.isEmpty() && externalRegistriesLoginDetails!= null && !externalRegistriesLoginDetails.isEmpty()){
+            for (Entry<String, String> loginDetail: externalRegistries.entrySet()) {
+                String registry = loginDetail.getKey();
+                String username = loginDetail.getValue();
+
+                if(registry == null || username == null){
+                    throw new IllegalArgumentException(String.format("Either registry {%s} or username {%s} was null ", registry, username));
+                }
+
+                if(!externalRegistriesLoginDetails.containsKey(username)){
+                    throw new IllegalArgumentException(String.format("registry {%s} and username {%s} supplied but no password in DOCKER_EXTERNAL_REGISTRY_LOGIN_DETAILS", registry, username));
+                }
+
+                String password = externalRegistriesLoginDetails.get(username);
+                if(password == null) {
+                    throw new IllegalArgumentException(String.format("registry {%s} and username {%s} supplied but password in DOCKER_EXTERNAL_REGISTRY_LOGIN_DETAILS is null", registry, username));
+                }
+
+                runDockerCommand(String.format("login  -e \"fake\" -u %s -p %s %s", username, password, registry));
```

Comment:
Meh. Change it to `fake@example.org` then, to make it clear it's an email address, and an unusable one, at thet.